.gitignore

```diff
@@ -0,0 +1,19 @@
+# Ignore temporary files and directories
+.ci-temp/
+.DS_Store
+
+# Ignore IntelliJ IDEA project files
+.idea/
+AbbreviationAsWordInName/.idea/
+SummaryJavadoc/.idea/
+
+# Ignore generated files
+extractor/src/test/resources/generated-config-all-in-one.xml
```

Comment:
Can we generate such files in .ci-temp ?
Or in "build" folder where all classes are generated, that supposed to be ignored by all means

Reply:
```
piyushkumarsadangi@piyushs-MacBook-Air test-configs % cd .ci-temp 
piyushkumarsadangi@piyushs-MacBook-Air .ci-temp % ls
checkstyle                      generated-config-example1.xml
piyushkumarsadangi@piyushs-MacBook-Air .ci-temp % cd 
```
We can. And, this requires just change in the output file path in test class. even for the build, i guess all i have to do is change the output file path. Is that what you are expecting? Changing the output file path to ci-temp. And, make sure we have ci-temp ready before running any test or Main.java (later on)?

Reply:
```
 @Test
    public void testGeneratedConfigMatchesExpectedExample1() throws Exception {
        String exampleFilePath = "../.ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example1.java";
        String templateFilePath = "src/main/resources/config-template-treewalker.xml";
        String outputFilePath = "../.ci-temp/generated-config-example1.xml";
        String expectedFilePath = "src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/configs/config-example1-expected.xml";

        ConfigSerializer.serializeConfig(exampleFilePath, templateFilePath, outputFilePath);

        String generatedContent = new String(Files.readAllBytes(Path.of(outputFilePath)));
        String expectedContent = new String(Files.readAllBytes(Path.of(expectedFilePath)));

        // Sort the properties in both the generated and expected content
        String sortedGeneratedContent = ConfigSerializer.sortProperties(generatedContent);
        String sortedExpectedContent = ConfigSerializer.sortProperties(expectedContent);

        assertEquals(sortedExpectedContent.trim(), sortedGeneratedContent.trim(), "The generated configuration does not match the expected output.");
    }
```

Reply:
Gradle created folder "build" for all output of compilation and any other task.
https://stackoverflow.com/q/37512772/1015848

Let's reuse "build" or just keep result content of file as string and load expected file to string, and compare two strings in assert. But use Truth library assert, it will show diff in good way.

Reply:
![Uploading Screenshot 2024-06-14 at 12.39.51 AM.png…]()

Reply:
```
% ./gradlew clean build


> Task :compileJava
Note: /Users/piyushkumarsadangi/Development/test-configs/extractor/src/main/java/com/example/extractor/ConfigSerializer.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :setupCheckstyleRepo
Checkstyle repository already cloned.

> Task :test

MainTest > testGeneratedConfigMatchesExpectedAllInOne() STANDARD_OUT
    Generating all-in-one configuration...
    Loading configuration from example file: ../.ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example1.java
    Loading configuration from example file: ../.ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example2.java
    Loading configuration from example file: ../.ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example3.java
    Loading configuration from example file: ../.ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example4.java
    Loading configuration from example file: ../.ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example5.java
    Loading configuration from example file: ../.ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example6.java
    Loading configuration from example file: ../.ci-temp/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/Example7.java

MainTest > testGeneratedConfigMatchesExpectedAllInOne() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample1() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample1() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample2() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample2() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample3() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample3() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample4() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample4() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample5() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample5() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample6() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample6() PASSED

MainTest > testGeneratedConfigMatchesExpectedExample7() STANDARD_OUT
    Loading configuration from example file...

MainTest > testGeneratedConfigMatchesExpectedExample7() PASSED

MainTest > testParseConfigFromHeader() PASSED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.6/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 11 executed

```

Reply:
@Romani, are the changes good? right, now on doing clean build, i can see all the generated files in the build section. and, making comparison on string basis.

Reply:
I still see
```
# Ignore generated files
extractor/src/....
```
Please remove